rest_framework_simplejwt/tokens.py

```diff
@@ -202,6 +202,14 @@ def for_user(cls: Type[T], user: AuthUser) -> T:
         Returns an authorization token for the given user that will be provided
         after authenticating the user's credentials.
         """
+        # Prevent incorrect usage of Token.for_user (when creating tokens manually)
+        # see https://github.com/jazzband/djangorestframework-simplejwt/issues/779
+        if not api_settings.USER_AUTHENTICATION_RULE(user):
+            raise TokenError(_("Token is invalid or expired"))
+        return cls.for_validated_user(user)
+
+    @classmethod
+    def for_validated_user(cls: Type[T], user: AuthUser) -> T:
         user_id = getattr(user, api_settings.USER_ID_FIELD)
```

Comment:
I would like the method still be documented, with a warning explaining what it does/does not and what to use instead.

For example,
```suggestion
    def for_validated_user(cls: Type[T], user: AuthUser) -> T:
        """
        Returns an authorization token for the given user.
        
        This DOES NOT check if the provided user validates the ``USER_AUTHENTICATION_RULE``.
        Use :meth:`for_user` to have the user validated.
        """
        user_id = getattr(user, api_settings.USER_ID_FIELD)
```